src/App.tsx

```diff
@@ -8,53 +10,175 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { useEffect, useMemo, useState } from 'react';
+import * as servicesUsers from './api/users';
+import * as servicesPosts from './api/posts';
+import * as servicesComments from './api/comments';
+import { Post } from './types/Post';
+import { Comment } from './types/Comment';
 
-export const App = () => (
-  <main className="section">
-    <div className="container">
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState([]);
+  const [isUser, setIsUser] = useState(false);
+  const [userId, setUserId] = useState<number | null>(null);
+  const [posts, setPosts] = useState([]);
+  const [comments, setComments] = useState([]);
+  const [currentPostId, setCurrentPostId] = useState(0);
+  const [openPostId, setOpenPostId] = useState<number | null>(null);
+  const [isWritePostButton, setIsWritePostButton] = useState(false);
+  const [loading, setLoading] = useState('');
+  const [isError, setIsError] = useState(false);
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    setLoading('users');
+    servicesUsers
+      .getUsers()
+      .then(res => {
+        setUsers(res);
+        setIsError(false);
+      })
+      .catch(() => setIsError(true))
+      .finally(() => setLoading(''));
+  }, []);
 
-              <Loader />
+  useEffect(() => {
+    if (userId) {
+    setLoading('posts');
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+    servicesPosts.getPosts(userId)
+      .then(res => {
+        setPosts(res);
+        setIsError(false);
+      })
+      .catch(() => setIsError(true))
+      .finally(() => setLoading(''));
+    }
+}, [userId]);
+
+  const getCommentsByPostId = (postId: number) => {
+    setLoading('comments');
+    setCurrentPostId(postId);
+
+    return servicesComments
+      .getComments(postId)
+      .then(res => {
+        setComments(res);
+        setIsError(false);
+      })
+      .catch(() => setIsError(true))
+      .finally(() => setLoading(''));
```

Comment:
The `.catch` block should be placed before the `.then` block in the `getCommentsByPostId` function to ensure proper error handling. This will allow the function to handle errors before attempting to process the response.